internal/component/database_observability/postgres/component.go

```diff
@@ -284,19 +284,19 @@ func (c *Component) Update(args component.Arguments) error {
 		return nil
 	}
 
-	var systemID, systemIP, systemPort, engineVersion string
+	var systemID, systemIP, systemPort, engineVersion sql.NullString
 	if err := rs.Scan(&systemID, &systemIP, &systemPort, &engineVersion); err != nil {
 		c.reportError("failed to scan engine version", err)
 		return nil
 	}
 
-	systemID = fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%s:%s:%s", systemID, systemIP, systemPort))))
+	generatedSystemID := fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%s:%s:%s", systemID.String, systemIP.String, systemPort.String))))
```

Comment:
didn't really read into it, but will everything depending on this generated system ID be OK when the ID and Port are both null (empty string)? we won't have issues with the same generated ID?

Reply:
The reason we use `systemID` + `IP` + `port` is that clustered postgres instances share the `systemID` (not unique per instance within a cluster), so the `IP` and `port` add uniqueness. 

Taking this away means that clustered postgres instances may share a `server_id` - with all the issues that causes around aggregation. 
However this only will happen when a unix domain socket is used - whereas today the component fails.

I don't think this is a long term solution, but it stops the worst thing happening where all UDP connections fail